### DISCUSSION TITLE
Model error during simulation

### DISCUSSION THREAD

**Discussion by xylhal** [the-anylogic-company/Alpyne#30] [Q&A]
Hello, 

I'm trying to train an RL algorithm with a simulation model and I'm getting the following model error that randomly pops up during training. 

To further debug, I have removed all RL components, fixed the initial seed and hard-coded the actions taken. Essentially, I am now just running a fixed number of episodes with the same action being taken every single time and verified that the consecutive states are the same for every episode across all states. However, what's happening is that the error below will randomly occur after after running for X amount of episodes successfully. This specifically happens during the env.step() part of the process.

![Screen Shot 2022-12-07 at 6 35 48 PM](https://user-images.githubusercontent.com/114612681/206342524-b6504535-1b0e-4bd5-839a-03bccbab50fe.png)

I've also attached the log files of a few runs but I couldn't identify the possible source of error from the error stack.

What's weird to me is that the simulation runs fine in Anylogic and the RL part runs successfully for some number of episode and but fails suddenly despite all the episodes being the same. 

Would you have any idea as to what's the possible cause of this error? 

[alpyne (1).log](https://github.com/t-wolfeadam/Alpyne/files/10181615/alpyne.1.log)
[alpyne.log](https://github.com/t-wolfeadam/Alpyne/files/10181616/alpyne.log)
[alpyne.log](https://github.com/t-wolfeadam/Alpyne/files/10181618/alpyne.log)

**Comment by mjhoover1** [comment]
Any update on this issue? Thanks.
